### PR TITLE
feat: add message signing tutorial

### PR DESCRIPTION
**Pull request by beguene**
## Description

I added a tutorial for Arbitrary Message Signing https://github.com/hirosystems/stacks-wallet-web/pull/2350

## Checklist

- [x] [Conventional commits were used](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] New links to files and images were verified
- [ ] For fixes/refactors: all existing references were identified and replaced
- [ ] [Style guide](https://developers.google.com/style) was reviewed and applied
- [x] Clear code samples were provided
- [x] People were tagged for review

@markmhx

### CODE REVIEW

File: docs/build-apps/message-signing.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
+---
+id: message-signing
+title: Signing messages
+---
+
+This guide explains how to prompt users to sign a message.
+
+The user will be prompted a popup from the Hiro web wallet showing the message you would like him to sign.
```
[the remark this answers is not in the document]
Comment (markmhx) [reply docs/build-apps/message-signing.md:8]: "..like them to sign"

File: docs/build-apps/message-signing.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+openSignatureRequestPopup({
+  network: mainnet,
+  // other relevant options
+});
+```
+
+## Usage in React Apps
+
+Import the `useConnect` from the [`connect-react`](https://github.com/hirosystems/connect) package to message signing more seamlessly into React apps.
```
[the remark this answers is not in the document]
Comment (markmhx) [reply docs/build-apps/message-signing.md:133]: "...package to sign messages more seamlessly with React apps."

File: docs/build-apps/message-signing.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+interface SignatureData {
+  /* Hex encoded DER signature */
+  signature: string;
+  /* Hex encoded private string taken from privateKey */
+  publicKey: string;
+}
+```
+## StacksProvider injected variable
+
+When users have the [Hiro Wallet for Web](https://www.hiro.so/wallet/install-web) browser extension installed, the extension will inject a global `StacksProvider` variable into the JavaScript context of your web app. This allows your JavaScript code to hook into the extension, and make authentication, transaction requests and signature requests. `@stacks/connect` automatically detects and uses this global variable for you.
```
Comment (markmhx) [inline docs/build-apps/message-signing.md:199]: "When users have the Hiro Wallet extension installed..."

"..and make authentication, transaction and signature requests."

File: docs/build-apps/message-signing.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (beguene) [reply docs/build-apps/message-signing.md]: I use the word 'helper' instead of 'module' since it's a react kind of function. Let me know if that is fine.

**Review by markmhx — APPROVED** [review]

**Pull request merged**